### PR TITLE
Unify the logics to check eager mode

### PR DESCRIPTION
**Pull request by JackCaoG**
Prior to my eager mode changes, eager mode is a debug only mode that's controlled by the `XLA_USE_EAGER_DEBUG_MODE ` env var, now since we want to make it more offical I want to control it explictly with the API instead of the env var. @aws-rhsoln let me know what you think. Eventually I want to deprecate the env var(at least not call it debug mode).

### CODE REVIEW

**Comment by aws-rhsoln** [comment]
Do you think its valuable to add a message letting the users know that eager mode run without torch.compile is going to be slow, can provide some basic reasoning as to why that would be the case. 
Need to let users know that graph based mode is still going to be faster (be it using compile or lazy eval).

**Comment by JackCaoG** [comment]
@aws-rhsoln check my doc at https://github.com/pytorch/xla/pull/7700, I am still thinking about how to message users better to let them know eager mode exists. but yea I agree that we have to make it clear that compile is the way to go for perfomrance.

**Comment by JackCaoG** [comment]
@aws-rhsoln for some reason I can't make you a reviewer, do you mind also taking a look at this pr?

File: torch_xla/__init__.py [hunk 1]
```diff
@@ -243,6 +243,10 @@ def _init_xla_lazy_backend():
   plugins.use_dynamic_plugins()
   plugins.register_installed_plugins()
 
+if os.getenv('XLA_USE_EAGER_DEBUG_MODE', '0') == '1':
```
Comment (aws-rhsoln) [inline torch_xla/__init__.py:246]: Is this just for backward compatibility, since we already have the api now?

File: torch_xla/csrc/tensor.cpp [hunk 2]
```diff
@@ -86,8 +86,7 @@ XLATensorPtr XLATensor::Create(
       XLATensor(std::move(ir_value), device, logical_element_type));
   XLAGraphExecutor* graph_executor = XLAGraphExecutor::Get();
   graph_executor->RegisterTensor(xtensor->data());
-  if ((UseEagerDebugMode() || graph_executor->UseEagerMode()) &&
-      !delay_eager_executation) {
+  if ((graph_executor->UseEagerMode()) && !delay_eager_executation) {
```
Comment (aws-rhsoln) [inline torch_xla/csrc/tensor.cpp:89]: Nit: Extra brackets around `graph_executor->UseEagerMode()`

**Review by aws-rhsoln — APPROVED** [review]
Just one nit, lgtm otherwise! Thanks!

**Review by alanwaketan — APPROVED** [review]
LGTM.

File: torch_xla/__init__.py (hunk 1 above)
Comment (JackCaoG) [reply torch_xla/__init__.py:246]: yea, it is mostly for backward compatibility. I think I can also tweak it and use it in test to test both eager and non-eager.

File: torch_xla/csrc/tensor.cpp (hunk 2 above)
Comment (JackCaoG) [reply torch_xla/csrc/tensor.cpp:89]: nice catch

**Pull request merged**